apps/sms/test/marionette/composer_test.js

```diff
@@ -42,6 +42,57 @@ marionette('Messages Composer', function() {
     );
   });
 
+  suite('Preserve message input while navigating', function() {
+    var composer, threadList, thread;
+    var message = 'test message';
+
+    function waitForThreadList() {
+      client.helper.waitForElement(threadList.mmsThread);
+    }
+
+    function createMMSThread() {
+      threadList.navigateToComposer();
+      messagesApp.addRecipient('a@b.c');
+      messagesApp.addRecipient('s@p.c');
+      composer.messageInput.sendKeys('MMS thread.');
+      messagesApp.send();
+      messagesApp.performHeaderAction();
```

Comment:
nit: maybe move ```messagesApp.performHeaderAction();``` to setup as separate statement, so that name ```createMMSThread``` corresponds to what it actually does? Now it's more like ```createMMSThreadAndNavigateToThreadList``` :)